granadilla/forms.py

```diff
@@ -48,6 +49,116 @@ class Meta:
         model = LdapContact
         exclude = ("dn", "photo", "full_name")
 
+class LdapDeviceForm(forms.ModelForm):
+
+    password2 = forms.CharField(widget=forms.PasswordInput, label = "Re-Password")
+
+    def __init__(self, *args, **kwargs):
+        if 'base_dn' in kwargs:
+            self.__base_dn = kwargs.pop('base_dn')
+        else:
+            self.__base_dn = None
+        self.__username = kwargs.pop('username')
```

Comment:
You could probably use a simple protected attribute instead, i.e ``self._base_dn``.
Using private attributes (starting with two underscores) involves name mangling and stuff.